minio/api.py

```diff
@@ -1858,23 +1863,15 @@ def _url_open(self, method, bucket_name=None, object_name=None,
             self._delete_bucket_region(bucket_name)
 
             # Populate response_error with error response.
-            response_error = ResponseError(response)
+            response_error = ResponseError(response, method).get_exception()
 
             # In case we did not preload_content, we need to release
             # the connection:
             if not preload_content:
                 response.release_conn()
 
-            if method == 'HEAD':
-                raise response_error.head(bucket_name, object_name)
-            elif method == 'GET':
-                raise response_error.get(bucket_name, object_name)
-            elif method == 'POST':
-                raise response_error.post(bucket_name, object_name)
-            elif method == 'PUT':
-                raise response_error.put(bucket_name, object_name)
-            elif method == 'DELETE':
-                raise response_error.delete(bucket_name, object_name)
+            if method in self.supported_methods:
```

Comment:
The whole code flow of creating a `ResponseError(response)` just seems weird in general...